src/python/zquantum/core/interfaces/optimizer_test.py

```diff
@@ -17,41 +18,83 @@ def sum_x_squared(x):
 
 
 class OptimizerTests(object):
-    # To run tests with this base class, the following variables need to be properly initialized in the child class:
-    # self.optimizers
-
-    def test_optimizer_succeeds_with_optimizing_rosenbrock_function(self):
-        for optimizer in self.optimizers:
-            cost_function = FunctionWithGradient(rosenbrock_function, finite_differences_gradient(rosenbrock_function))
-
-            results = optimizer.minimize(cost_function, initial_params=np.array([0, 0]))
-            self.assertAlmostEqual(results.opt_value, 0, places=4)
-            self.assertAlmostEqual(results.opt_params[0], 1, places=3)
-            self.assertAlmostEqual(results.opt_params[1], 1, places=3)
-
-            self.assertIn("nfev", results.keys())
-            self.assertIn("nit", results.keys())
-            self.assertIn("opt_value", results.keys())
-            self.assertIn("opt_params", results.keys())
-            self.assertIn("history", results.keys())
-
-    def test_optimizer_succeeds_with_optimizing_sum_of_squares_function(self):
-        for optimizer in self.optimizers:
-            cost_function = FunctionWithGradient(sum_x_squared, finite_differences_gradient(sum_x_squared))
-
-            results = optimizer.minimize(
-                cost_function, initial_params=np.array([1, -1])
-            )
-            self.assertAlmostEqual(results.opt_value, 0, places=5)
-            self.assertAlmostEqual(results.opt_params[0], 0, places=4)
-            self.assertAlmostEqual(results.opt_params[1], 0, places=4)
-
-            self.assertIn("nfev", results.keys())
-            self.assertIn("nit", results.keys())
-            self.assertIn("opt_value", results.keys())
-            self.assertIn("opt_params", results.keys())
-            self.assertIn("history", results.keys())
+    # To run tests with this class, Inherit it, and in the same module
+    # define a fixture with name optimizer that returns an optimizer.
```

Comment:
Maybe it would be a worthwhile to add an example of how to do that in the comment?

Reply:
Maybe we should just add a reference to one of tests in optimizers? Wdyt?

Reply:
I think it's better to add a minimal example rather than reference an external repo.
Though on the other hand, they are all so connected that it's probably fine as well.
Whatever you think is best.